packages/shorebird_cli/lib/src/commands/preview_command.dart

```diff
@@ -222,4 +260,24 @@ extension on String {
     final file = File(path)..createSync(recursive: true);
     await response.pipe(file.openWrite());
   }
+
+  Future<void> downloadAndExtract(String path) async {
+    final uri = Uri.parse(this);
+    final client = HttpClient();
+    final request = await client.getUrl(uri);
+    final response = await request.close();
+    if (response.statusCode != 200) {
+      throw Exception('Failed to download artifact at $this');
+    }
+    final tempDir = Directory.systemTemp.createTempSync();
```

Comment:
You probably want someone to pass you a FileSystem too?

Reply:
Just staying consistent with other code but we can refactor to use a FS wrapper in future PRs.